components/settings/SettingsLanguage.vue

```diff
@@ -1,16 +1,22 @@
 <script lang="ts" setup>
 import type { ComputedRef } from 'vue'
+import ISO6391 from 'iso-639-1'
 import type { LocaleObject } from '#i18n'
 
 const userSettings = useUserSettings()
 
 const { locales } = useI18n() as { locales: ComputedRef<LocaleObject[]> }
+
+// Find English name from language code (e.g. 'ja-JP' -> 'Japanese')
+function getEnglishName(langCode: string): string {
+  return ISO6391.getName(langCode.split('-')[0])
```

Comment:
The `ISO6391` library has language code data like `en` instead of `en-US` so I had to use the first part of the given language code here.